atc/db/worker_resource_cache.go

```diff
@@ -70,54 +81,61 @@ func (workerResourceCache WorkerResourceCache) FindOrCreate(tx Tx) (*UsedWorkerR
 	}, nil
 }
 
+// Find looks for a worker resource cache by resource cache id and worker name. It may find multiple
+// worker resource caches, just return the first valid one.
+// Note: Find doesn't consider workerResourceCache.SourceWorker as a search condition.
 func (workerResourceCache WorkerResourceCache) Find(runner sq.Runner) (*UsedWorkerResourceCache, bool, error) {
-	baseResourceType := workerResourceCache.ResourceCache.BaseResourceType()
-	usedWorkerBaseResourceType, found, err := WorkerBaseResourceType{
-		Name:       baseResourceType.Name,
-		WorkerName: workerResourceCache.WorkerName,
-	}.Find(runner)
-	if err != nil {
-		return nil, false, err
-	}
-
-	if !found {
-		return nil, false, nil
-	}
-
-	id, found, err := workerResourceCache.find(runner, usedWorkerBaseResourceType)
+	ids, workerBasedResourceTypeIds, found, err := workerResourceCache.find(runner, workerResourceCache.WorkerName)
 	if err != nil {
 		return nil, false, err
 	}
 
 	if found {
-		return &UsedWorkerResourceCache{
-			ID: id,
-		}, true, nil
+		// Verify worker base resource type
+		for i, workerBasedResourceTypeId := range workerBasedResourceTypeIds {
+			_, foundBrt, err := WorkerBaseResourceType{}.FindById(runner, workerBasedResourceTypeId)
```

Comment:
I don't think we need to verify the worker base resource type here, unless I'm missing something? But from what I can tell, we grab the `worker_base_resource_type_id` from the `workerResourceCache.find` query right before. I think we can trust that `worker_base_resource_type_id` is valid because that column has a foreign key to the `worker_base_resource_types` table with an `ON DELETE CASCADE`.